src/AppComponents/PizzaForm.jsx

```diff
@@ -0,0 +1,261 @@
+import { customAlphabet, nanoid } from 'nanoid';
+import React, { useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import shortid from 'shortid';
+import * as yup from 'yup';
+
+function PizzaForm(props) {
+  const history = useHistory();
+  const [pizzaOptions, setPizzaOptions] = useState({
+    size: ['Small', 'Medium', 'Large', 'Extra Large'],
+    sauce: ['Original Red', 'Garlic Ranch', 'BBQ Sauce', 'Spinach Alfredo'],
+    toppings: [
+      'Pepperoni',
+      'Sausage',
+      'Canadian Bacon',
+      'Spicy Italian Sausage',
+      'Garlic Chicken',
+      'Onions',
+      'Green Pepper',
+      'Diced Tomatos',
+      'Black Olives',
+      'Roasted Garlic',
+      'Artichoke Hearts',
+      'Three Cheese',
+      'Pineapple',
+      'Extra Cheese',
+    ],
+    substitute: ['Gluten Free Crust'],
+  });
+
+  const [order, setOrder] = useState({
+    name: '',
+    size: '',
+    sauce: '',
+    toppings: [],
+    substitute: '',
+    instructions: '',
+  });
+
+  const [isSubmitEnabled, setSubmitEnabled] = useState(false);
+
+  const formSchema = yup.object().shape({
+    name: yup.string().required('Name for order is required'),
+    size: yup.string().required('Size is required'),
+    sauce: yup.string(),
+    toppings: yup.array().max(10, "Can't have more than 10 toppings"),
+    substitute: yup.string(),
+    instructions: yup.string(),
+  });
+
+  useEffect(() => {
+    formSchema.isValid(order).then((valid) => {
+      setSubmitEnabled(valid);
+    });
+  }, [order]);
+
+  const changeInput = (evt) => {
+    if (evt.target.type !== 'radio' || evt.target.type !== 'checkbox') {
+      setOrder({ ...order, [evt.target.name]: evt.target.value });
+    }
+
+    if (evt.target.type === 'radio') {
```

Comment:
this if block seems the same as the one on line 58, why have both?

Reply:
I had some previous code that made it necessary I think and I just forgot to refactor